File: src/main/java/com/example/demo/model/AccessLog.java
package com.example.demo.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

/**
 * HTTP access log entity.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AccessLog {
    private Long id;
    @Builder.Default
    private OffsetDateTime createdAt = OffsetDateTime.now();
    private String requestId;
    private String method;
    private String path;
    private String query;
    private Integer status;
    private Long durationMs;
    private String remoteIp;
    private String userAgent;
    private String referer;
    private String username;
    private Long requestBytes;
    private Long responseBytes;
}

File: src/main/kotlin/com/example/demo/model/AccessLog.kt
package com.example.demo.model

import java.time.OffsetDateTime

/**
 * HTTPアクセスログのエンティティ
 */
data class AccessLog(
    val id: Long? = null,
    val createdAt: OffsetDateTime = OffsetDateTime.now(),
    val requestId: String,
    val method: String,
    val path: String,
    val query: String? = null,
    val status: Int,
    val durationMs: Long? = null,
    val remoteIp: String? = null,
    val userAgent: String? = null,
    val referer: String? = null,
    val username: String? = null,
    val requestBytes: Long? = null,
    val responseBytes: Long? = null,
)
